### PR TITLE
Serve canvas background color / feature selection via proper QgsProject API

### PR DESCRIPTION
**Pull request by nirvn**
## Description

Code cleanup to avoid using a bunch of writeEntry / readNumEntry across the code, and serve a stable API setter/getter with test cases for 3rd parties to make use of.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [x] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### CODE REVIEW

**Comment by 3nids** [comment]
really nice, wanted to do this for a while.
But what about going further and make a generic method in `QgsProject::const QString &scope, const QString &key, const QColor &color ) SIP_PYNAME( writeColorEntry );`

**Comment by nirvn** [comment]
@3nids , identified this API gap while fixing QField's canvas background color situation :)

**Comment by nyalldawson** [comment]
Much cleaner! One last request - can you test and ensure that the new signals aren't emitted when setting a color equal to the existing one?

**Comment by nirvn** [comment]
@nyalldawson , done.

File: src/gui/qgsmapcanvas.cpp [hunk 1]
```diff
@@ -1823,7 +1823,14 @@ void QgsMapCanvas::setSelectionColor( const QColor &color )
     return;
 
   mSettings.setSelectionColor( color );
-
+  QgsDebugMsg( "XXXXXXXXXXXXXXXXX" );
```
Comment (nyalldawson) [inline src/gui/qgsmapcanvas.cpp]: ?

Comment (nirvn) [reply src/gui/qgsmapcanvas.cpp]: Removed. It's my old fashioned / messy debugging style :)

**Pull request merged**